src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy

```diff
@@ -942,31 +946,52 @@ class OpenShiftDSL implements Serializable {
     }
 
     public boolean verifyService(String svcName) {
+
         OpenShiftResourceSelector svcSel = this.selector("svc", svcName);
         String ip = (String)svcSel.object().get("spec").get("clusterIP");
         String port = (String)svcSel.object().get("spec").get("ports").get(0).get("port");
+        String targetPort = (String)svcSel.object().get("spec").get("ports").get(0).get("targetPort");
+
         int i = 0;
         while (i < 5) {
             i++;
-            InetSocketAddress address = new InetSocketAddress(ip, Integer.parseInt(port));
-            Socket socket = null;
-            try {
-                socket = new Socket();
-                socket.connect(address, 2500);
-                logToTaskListener("Connected to service " + svcName + " via IP " + ip + " and port " + port);
-                return true;
-            } catch (IOException e) {
-                logToTaskListener("IOException " + e.getMessage() + " connecting to service " + svcName);
-                LOGGER.log(Level.FINE, "verifyService", e);
-                try {
-                    Thread.sleep(2500);
-                } catch (InterruptedException e1) {
+            if (ip == "None"){
```

Comment:
`oc` is known to change  ... we might always get "None"

how comfortable are you constructing a regular expression check or some other validation that determines if it is a valid ipv4 / ipv6 address ?

Reply:
@gabemontero did you mean:
oc is known to change ... we might **NOT**  always get "None"

I am not sure about this, because setting None as a cluster IP is the explicit way to define a headless service as stated in [1].

If it is not the case, then @waveywaves, the check needs to be simplify. Could be useful to use Apache commons-validator which has an InetAddressValidator.isInetAddress method.


If we don't have this dependency, which seems to be the case, at least, we need to extract constants of the regex.

[1] https://kubernetes.io/docs/concepts/services-networking/service/#headless-services